### PR TITLE
Klientctl start healthcheck

### PR DESCRIPTION
**Pull request by leeola**
Klient can "fail to start" even if the users internet is down, or
other external issues. By health checking, we attempt to give them
accurate information of possible causes for klient not starting.

Also, for debugging, we health check before even starting klient - just
to have a record of if the user tried to start klient before their
internet was fully up _(mainly for debugging purposes)_.

### CODE REVIEW

File: go/src/koding/klientctl/config/config.go [hunk 1]
```diff
@@ -82,6 +82,11 @@ var (
 	// KontrolURL is overwritten during deploy via linker flag.
 	KontrolURL = "https://koding.com/kontrol/kite"
 
+	// TunnelKiteAddress is the address that koding's tunnel service is run on.
+	//
+	// This is overwritten during deploy via linker flag.
+	TunnelKiteAddress = "http://t.koding.com/kite"
```
Comment (rjeczalik) [inline go/src/koding/klientctl/config/config.go]: TunnelKite is t.koding.com for production and dev-t.koding.com for development.

Could you add this url to koding/config/config.json and read it from there? We're going to move all the hardcoded urls and settings to that package either way, the less new ones we add in the meantime the better.

File: go/src/koding/klientctl/status.go [hunk 2]
```diff
@@ -158,152 +216,184 @@ func (c *HealthChecker) CheckLocal() error {
...
-	if err != nil {
-		return ErrHealthUnexpectedResponse{Message: fmt.Sprintf(
-			"Error reading response from %s: '%s'",
-			c.RemoteKiteAddress, p,
-		)}
+	if err := c.LocalRequirements(); err != nil {
+		switch err.(type) {
+		// Ignore dialing or bad klient http responses for CheckAllExceptRunning.
+		case ErrHealthNoHTTPReponse:
+		case ErrHealthDialFailed:
```
Comment (rjeczalik) [inline go/src/koding/klientctl/status.go]:
```
case ErrHealthNoHTTPReponse, ErrHealthDialFailed:
```

File: scripts/deploy-kd.sh [hunk 3]
```diff
@@ -28,10 +28,15 @@ if [[ "$CHANNEL" == "development" ]]; then
 	KONTROL_URL="https://sandbox.koding.com/kontrol/kite"
 fi
 
+TUNNEL_URL="http://t.koding.com/kite"
```
Comment (rjeczalik) [inline scripts/deploy-kd.sh]: Could you please also edit klientctl/build.sh script?

File: go/src/koding/klientctl/config/config.go (hunk 1 above)
Comment (rjeczalik) [inline go/src/koding/klientctl/config/config.go]: I was too fast, now I noticed we overwrite the string during compile. Well, the long term plan is to get rid of injecting configuration string during compile in order to have only 1 klient channel, not 4 (less things to compile and deploy).

For now we must live with the injection during compile time.

File: scripts/deploy-kd.sh (hunk 3 above)
Comment (leeola) [inline scripts/deploy-kd.sh]: @rjeczalik done via https://github.com/koding/koding/pull/8887/commits/e8f02d579e5eec24074cde42d5be1d184671e198

File: go/src/koding/klientctl/build.sh [hunk 4]
```diff
@@ -23,13 +23,18 @@ if [[ "$CHANNEL" == "development" ]]; then
...
+fi
+
 if [[ -z "$VERSION" ]]; then
 	VERSION=$(curl -sSL https://koding-kd.s3.amazonaws.com/${CHANNEL}/latest-version.txt)
 	let VERSION++
 fi
 
 kd_build() {
-	go build -v -ldflags "-X koding/klientctl/config.Version=$VERSION -X koding/klientctl/config.SegmentKey=$KD_SEGMENTIO_KEY -X koding/klientctl/config.Environment=$CHANNEL -X koding/klientctl/config.KontrolURL=$KONTROL_URL" koding/klientctl
+	go build -v -ldflags "-X koding/klientctl/config.Version=$VERSION -X koding/klientctl/config.SegmentKey=$KD_SEGMENTIO_KEY -X koding/klientctl/config.Environment=$CHANNEL -X koding/klientctl/config.TunnelKiteAddress $TUNNEL_URL --X koding/klientctl/config.KontrolURL=$KONTROL_URL" koding/klientctl
```
Comment (rjeczalik) [inline go/src/koding/klientctl/build.sh]: typo `--X koding/klientctl/config.KontrolURL` - it needs to be `-X`

Comment (leeola) [inline go/src/koding/klientctl/build.sh]: Nice catch, apologies. Fixed

**Pull request merged**